### PR TITLE
fix: [BREAKING] Use default name for components with setup script and nameless compon…

### PR DESCRIPTION
**Pull request by khalilst**
This PR in the following of issue #806 replaces the anonymous naming with the default component name:

So the following component's stub name will be `script-setup-stub`:
```js
//filename: ScriptSetup.vue
<script setup>
...
</script>
```

Also, the same approach works on nameless components.
The test coverage applied for this fix.

### CODE REVIEW

**Review by cexbrayat — APPROVED** [review]
Thanks @khalilst for the PR!

LGTM, but I'd like another pair of eyes before merging this.

**Comment by cexbrayat** [comment]
@lmiller1990 @xanf @afontcu I added you as reviewers, as I don't know if that's too breaking or OK to merge. What do you think?

**Comment by khalilst** [comment]
@cexbrayat 
My pleasure (^_^)
Sure, you need to check more. Just for your consideration, I've applied it in a local test env and everything works fine.

**Review by lmiller1990 — COMMENTED** [review]
I think this makes sense. Bit of a breaking change for snapshots, but this seems like a overall improvement.

**Pull request merged**
